QgisModelBaker/gui/workflow_wizard/workflow_wizard.py

```diff
@@ -448,7 +466,37 @@ def get_topping_file_model(self, id_list, log_panel):
 
         return topping_file_cache.model
 
-    def add_source(self, source):
+    def update_referecedata_cache_model(self, filter_models, type):
+        # updates the model and waits for the end
+        loop = QEventLoop()
+        self.ilireferencedatacache.model_refreshed.connect(lambda: loop.quit())
+        timer = QTimer()
+        timer.setSingleShot(True)
+        timer.timeout.connect(lambda: loop.quit())
+        timer.start(10000)
+        self.refresh_referencedata_cache(filter_models, type)
+        loop.exec()
+        return self.ilireferencedatacache.model
+
+    def refresh_referencedata_cache(self, filter_models, type):
+        self.ilireferencedatacache.base_configuration = self.base_config
+        self.ilireferencedatacache.filter_models = filter_models
+        self.ilireferencedatacache.type = type
+        self.ilireferencedatacache.refresh()
+
+    def _db_modelnames(self, configuration):
+        db_connector = db_utils.get_db_connector(configuration)
+        modelnames = list()
+        if db_connector:
+            if db_connector.db_or_schema_exists() and db_connector.metadata_exists():
+                db_models = db_connector.get_models()
+                for db_model in db_models:
+                    regex = re.compile(r"(?:\{[^\}]*\}|\s)")
```

Comment:
In Qt regex are heavy to build, so i would put it before the `for` loop and reuse it. Not sure if it is the same in python tought...